### PR TITLE
[lldb-dap] Refactoring JSONUtils to not use `g_dap` and instead passing in required arguments.

### PR DESCRIPTION
**Pull request by ashgti**
This is part of a larger refactor to remove the global `g_dap` variable.

### CODE REVIEW

File: lldb/tools/lldb-dap/JSONUtils.cpp [hunk 1]
```diff
@@ -1465,11 +1462,7 @@ CreateRunInTerminalReverseRequest(const llvm::json::Object &launch_request,
     llvm::StringRef key = envs.GetNameAtIndex(index);
     llvm::StringRef value = envs.GetValueAtIndex(index);
 
-    if (key.empty())
-      g_dap.SendOutput(OutputType::Stderr,
-                       "empty environment variable for value: \"" +
-                           value.str() + '\"');
-    else
```
Comment (walter-erquinigo) [inline lldb/tools/lldb-dap/JSONUtils.cpp:1472]: Why did you remove this?

**Review by walter-erquinigo — COMMENTED** [review]
Save one comment, everything else seems good

File: lldb/tools/lldb-dap/JSONUtils.cpp (hunk 1 above)
Comment (ashgti) [reply lldb/tools/lldb-dap/JSONUtils.cpp:1472]: I guess I wasn't really sure why this is a warning.

In lldb I can use `settings set target.env-vars FOO=` and there isn't any warnings when I run a process and having an empty env isn't an issue on mac or linux as far as I can tell. 

I also don't see any mentions of why this was added in github.com/llvm/llvm-project/pull/106919 I could move the warning into lldb-dap.cpp after we call this helper.

File: lldb/tools/lldb-dap/JSONUtils.cpp [hunk 2]
```diff
@@ -154,7 +154,7 @@ DecodeMemoryReference(llvm::StringRef memoryReference) {
 std::vector<std::string> GetStrings(const llvm::json::Object *obj,
                                     llvm::StringRef key) {
   std::vector<std::string> strs;
-  auto json_array = obj->getArray(key);
+  const auto *json_array = obj->getArray(key);
```
Comment (vogelsgesang) [inline lldb/tools/lldb-dap/JSONUtils.cpp:157]: ah, you are using clang-tidy? I always always get those warnings 😄

**Review by vogelsgesang — APPROVED** [review]

File: lldb/tools/lldb-dap/JSONUtils.cpp (hunk 1 above)
Comment (vogelsgesang) [reply lldb/tools/lldb-dap/JSONUtils.cpp:1472]: > In lldb I can use `settings set target.env-vars FOO=` [...]

Note that this warning warns about `key.empty()`, not `value.empty()`, i.e. for `=FOO` and not `FOO=`.

That being said, I am also not sure how much value this warning provides

Comment (ashgti) [reply lldb/tools/lldb-dap/JSONUtils.cpp:1472]: Ah, your right, I mixed that up. Trying to set an empty key in lldb produces:

```
(lldb) settings set target.env-vars =1
error: empty dictionary key
```

However, if I have `"env": ["=123"],` in my launch.json I do end up with the empty "" being set at runtime, at least on macOS, which is kinda of wild that it works.

```
(vsocde-repl) `expr (char*)getenv("")
(lldb) expr (char*)getenv("")
(char *) $0 = 0x000000016fdff65f "123"
```

File: lldb/tools/lldb-dap/JSONUtils.cpp (hunk 2 above)
Comment (ashgti) [reply lldb/tools/lldb-dap/JSONUtils.cpp:157]: Yea, trying to fix those as I work on refactoring `g_dap`.

**Review by labath — APPROVED** [review]

**Review by walter-erquinigo — APPROVED** [review]
